transcend/resource_data_silo.go

```diff
@@ -487,6 +492,25 @@ func resourceDataSilosUpdate(ctx context.Context, d *schema.ResourceData, m inte
 	var saasContext []byte
 	if d.Get("secret_context") != nil {
 		// Lookup the sombra URL to talk to
+		// var sombraUrlQuery struct {}
+		// if d.Get("sombraId") != nil {
+		// 	sombraUrlQuery = {
+		// 		Organization struct {
+		// 				Sombras struct {
+		// 					SombraId graphql.String `graphql:"sombraId"`
+		// 				}`graphql:"sombras"`
+		// 			}`graphql:"organization"`
```

Comment:
@dmattia My thinking here is to use a different query when someone enters a `sombraId` when one is specified verses just grabbing the primary sombra. While trying to write this query, I wasn't sure how to filter down to a specific sombra via `sombraId` since I didn't realize that the `sombras` query doesn't have filters.

Does this mean that I have to go back into the main repo and add a filter by id there and then use that filter here? Or is there another way to approach?

Reply:
Also for the update, I noticed that we were using `updateDataSilo` mutation verses the bulk update one. Poking around, it seems like there are TODOs in the code to deprecate the single update mutations in favor for the bulk update ones. Will these functions eventually be switched over to bulk updates too?

Reply:
Great questions!

> My thinking here is to use a different query when someone enters a sombraId when one is specified verses just grabbing the primary sombra

That sounds like a great idea

> Does this mean that I have to go back into the main repo and add a filter by id there and then use that filter here?

That makes sense. It would be possible to fetch all sombras and iterate over the results, but I think adding an optional filter by ID on our backend would be a better solution as you mentioned

> Will these functions eventually be switched over to bulk updates too?

It's pretty low priority in this repo, as Terraform has built-in rate-limiting to ensure our backend doesn't get overloaded. It would still be better to use our bulk upload endpoint from our system's resource usage POV, but it's not the worst thing to use the singular endpoint here just because terraform already protects us (and it's a _lot_ of work to do batching like that in tf, we'd almost need a new resource type potentially.

However, I do think one solution we discussed earlier that I'd love to eventually see would be for tf to use the bulk updates endpoint, but still just send one data silo's info at a time. It wouldn't impact scalability really, but would at least let us complete the deprecation and stop having to maintain multiple endpoints. Annalisa would be the expert on that work / if it's prioritized or not.

But for now, you don't _need_ to dig into that I don't think, but are very welcome if you'd like to. It should be safe to call just the singular endpoint in this PR, as the code already does.

Reply:
Thanks for answering! 

As for querying specific sombras by sombraId, I will go back into the main repo and add a new filter that can be used here.

As for the bulk updates, I'll just ignore that for now and focus on the task at hand. I was just curious!